### PR TITLE
es fix

### PR DESCRIPTION
**Pull request by NatalieEO**
**Description**
This was difficult to reproduce because it only happened if a default version had been set, which isn't required when using the API to publish. Anyway, just needed to follow the same steps that we do for bioworkflows. Added test.

But, indexing locally doesn't fail for me like it does on dev. I'm unclear whether this will resolve that issue, but I wasn't able to reproduce it if anyone has ideas.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-3846

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [ ] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] Check the Snyk dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java [hunk 1]
```diff
@@ -384,6 +381,19 @@ private static Entry removeIrrelevantProperties(final Entry entry) {
         return detachedVersions;
     }
 
+    public static Workflow detachWorkflow(Workflow detachedWorkflow, Workflow workflow) {
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java:384]: Should be private

**Review by coverbeck — APPROVED** [review]
> but I wasn't able to reproduce it if anyone has ideas.

If this doesn't work after deploy, you could try downloading the dev DB; sort of a hassle because we don't have Funkbot, but doable.

**Comment by kathy-t** [comment]
> But, indexing locally doesn't fail for me like it does on dev. I'm unclear whether this will resolve that issue, but I wasn't able to reproduce it if anyone has ideas.

I was able to reproduce it locally by registering a GitHub App tool, setting a default version for the tool, publishing, then indexing

**Review by denis-yuen — COMMENTED** [review]
unsure if being able to reproduce will change the fix

**Review by denis-yuen — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Review by garyluu — APPROVED** [review]

**Pull request merged**